lib/src/tasks/saucelab_tests/sauce_runner.dart

```diff
@@ -0,0 +1,233 @@
+/// A library for running Dart unit tests on Sauce Labs from start to finish,
+/// with automatic (and optional) management of the Pub server and Sauce Connect tunnel.
+library dart_dev.src.tasks.saucelab_tests.sauce_runner;
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:dart_dev/src/tasks/serve/api.dart';
+import 'package:dart_dev/util.dart';
+import 'package:http/http.dart';
+
+import 'sauce_unit_test_driver.dart';
+import 'platforms.dart';
+import 'sauce_api.dart';
+
+export 'platforms.dart';
+
+const String iframeRunnerPath = 'sauce_iframe_runner/sauce_iframe_runner.html';
+
+/// Runs the specified [tests] in Sauce Labs the specified [platforms], and returns a Future
+/// that completes with the test resuls after they've finished running.
+///
+/// Additional [options] can be used to start the test (see <https://wiki.saucelabs.com/display/DOCS/Test+Configuration+Options>).
+///
+/// If [pubServePort] is `0`, a Pub server will automatically be started in order to serve the tests.
+/// Otherwise, it will be assumed that a Pub server is already running on the specified port.
+///
+/// If [autoSauceConnect] is `true`, then a Sauce Connect tunnel will be automatically started,
+/// with the optional [tunnelIdentifier], if specified.
+/// If `false`, it will be assumed that Sauce Connect is already running (with [tunnelIdentifier], if specified).
+Future<Map> run(List<SauceTest> tests, List<SaucePlatform> platforms,
+    String sauceUserName, String sauceAccessKey,
+    {Map options: const {},
+    int pubServePort: 0,
+    bool autoSauceConnect: true,
+    String tunnelIdentifier}) async {
+  PubServeTask pubServeTask;
+  PubServeInfo serveInfo;
+  Process sauceConnect;
+  List<SauceTestHarness> testHarnesses;
+
+  try {
+    if (pubServePort == 0) {
+      reporter.log('Starting pub server...');
+      // Start `pub serve` on the `test` directory
+      pubServeTask = startPubServe(additionalArgs: ['test', '--force-poll']);
```

Comment:
i added this flag because i was running out of disk space when trying to run on skynet 2.0 (https://wf-skynet-hrd.appspot.com/results/smithy/520060/2) , the flag was based on a suggestion from someone from the test engineering group